Feature/top down pr

Lastdayends opened the pull request:
Top down routing flow to remove off grid issue.

Before merging into the master branch, we will do more test.

stevenmburns commented on PlaceRouteHierFlow/PnRDB/datatype.h:
```diff
@@ -266,6 +287,7 @@ struct block {
   //int SBidx;
   //int counterpart;
   // Block pin
+  vector<PowerNet> PowerNets;
```
@854768750 @Lastdayends Do we need this field (PowerNets in block) in the database? We have a similar one in hierNode.

854768750 replied:
PowerNets in block does not contain physical information (rect, point). Now it only has connection information (which pins are connected to VDD or VSS). PowerNets in hierNode has physical information.

Lastdayends commented:
@stevenmburns @854768750 @meghna09 

Can we merge this branch now?

Lastdayends commented:
@854768750 

For 2, as far as I remember, we fixed the pin name bug in some branch already.
For 3, I fixed it in last commit. Has been tested with vco_res design.

Yaguang

854768750 commented:
#415 solved some cases.
When the capacitor has two pins(MINUS and PLUS), it is fine.
When the capacitor has four pins(MINUS1, MINUS2, PLUS1, PLUS2). It has wrong pin names. 
There may be some other issues to the pin name.

Lastdayends commented:
@854768750 

I see. Thanks.

Yaguang
